tests/ci.js

```diff
@@ -1,181 +1,204 @@
 }
 
-shell.echo(filename + ': Uninstalling old elm-test...');
-shell.exec('npm remove --ignore-scripts=false --global ' + elmTest);
-
-shell.echo(filename + ': Installing elm-test...');
-shell.exec('npm link --ignore-scripts=false');
-
-var interfacePath = require('elmi-to-json').paths['elmi-to-json'];
-
-shell.echo(filename + ': Verifying elmi-to-json is installed...');
-var interfaceResult = spawn.sync(interfacePath, ['--help']);
-var interfaceExitCode = interfaceResult.status;
-
-if (interfaceExitCode !== 0) {
-  shell.echo(
-    filename +
-      ': Failed because `elmi-to-json` is present, but `elmi-to-json --help` returned with exit code ' +
-      interfaceExitCode
-  );
-  shell.echo(interfaceResult.stdout.toString());
-  shell.echo(interfaceResult.stderr.toString());
-  shell.exit(1);
+function assertTestFailure(runResult) {
+  const msg = 'Expected failure (exit code >= ' + resultFailureThreshold + '), but got ' + runResult.status;
+  assert.ok(runResult.status >= resultFailureThreshold, getDetailedMessage(msg, runResult));  
 }
 
-shell.exec('npm link --ignore-scripts=false');
-
-shell.echo(filename + ': Verifying installed elm-test version...');
-var versionRun = shell.exec(elmTest + ' --version');
-
-if (versionRun.code !== 0) {
-  shell.exec(
-    'echo Expected elm-test --version to exit with exit code 0, but it was ' +
-      versionRun.code
-  );
-  shell.exit(1);
-}
+describe('--help', () => {
+  it('Should exit indicating success', () => {
+    const runResult = execElmiToJSON(['--help']);
+    assertTestSuccess(runResult);
+  }).timeout(60000);
+
+  it('Should print the usage', () => {
+    const runResult = execElmiToJSON(['--help']);
+    // ensure we have a non-empty output
+    assert.ok(runResult.stdout.length > 0);
+  }).timeout(60000);
+});
 
-if (versionRun.stdout.trim() !== elmTestVersion) {
-  shell.exec(
-    'echo Expected elm-test --version to output ' +
-      elmTestVersion +
-      ', but it was ' +
-      versionRun.stdout.trim()
-  );
-  shell.exit(1);
-}
+// shell.exec('npm link --ignore-scripts=false');
+describe('--version', () => {
+  it('Should exit indicating success', () => {
+    const runResult = execElmTest(['--version']);
+    assertTestSuccess(runResult);
+  }).timeout(60000);
+
+  it('Should print the usage', () => {
+    const runResult = execElmTest(['--version']);
+    assert.strictEqual(elmTestVersion, runResult.stdout.trim());
+  }).timeout(60000);
+});
 
 /* Test examples */
 
-shell.echo('\n### Testing elm-test on example-application/');
-
-shell.cd('example-application');
-
-assertTestFailure();
-assertTestSuccess(path.join('tests', '*Pass*.elm'), false);
-assertTestFailure(path.join('tests', '*Fail*.elm'));
-
-shell.cd('../');
-
-shell.echo('\n### Testing elm-test on example-application-src/');
-
-shell.cd('example-application-src');
-
-assertTestSuccess('src');
+describe('Testing elm-test on an example application', () => {
+  before(() => {
+	  shell.pushd('example-application');
```

Comment:
Is it possible to pass the `cwd` option to `spawn.sync` instead? Then we should be able to get rid of `shelljs` altogether?

Reply:
This would be really cool. I love to reduce the number of deps we take. Feel free to leave this for another pull request though if you want.

Reply:
Currently, we're also using shelljs to enumerate test files (using shell.ls). What would be a good alternative to that (without introducing another dependency) ?

Reply:
Something like this should work:

```js
    const dir = path.join(fixturesDir, 'tests', 'Passing');
    fs.readdirSync(dir).forEach(function (testToRun) {
      const itsPath = path.join(dir, testToRun);
      const runResult = execElmTest([itsPath]);
      assertTestSuccess(runResult);
    });
```

Where `fs` comes from `const fs = require('fs')` which is the [native Node.js fs module](https://nodejs.org/api/fs.html).

Reply:
I've removed shelljs (cwd is now provided to spawn.sync, and we use fs.readdirSync() instead of shell.ls).